files/class-wp-filesystem-vip.php

```diff
@@ -25,6 +25,10 @@ public function __construct( $dependencies ) {
 		$this->method = 'vip';
 		$this->errors = new WP_Error();
 
+		if ( ! is_array( $dependencies ) || 2 !== count( $dependencies ) ) {
+			$dependencies = request_filesystem_credentials( site_url() );
```

Comment:
https://developer.wordpress.org/reference/functions/request_filesystem_credentials/

> bool|array True if no filesystem credentials are required, false if they are required but have not been provided, array of credentials if they are required and have been provided.

Do we need to protect against boolean returns?

Reply:
```sh
php -r 'error_reporting(-1); list($a, $b) = false;'
```

does not generate any warnings; we are probably safe.